### PR TITLE
macOS touch ups

### PR DESCRIPTION
**Pull request by Mrwerdo**
There are a few things in this pull request:

1. This reapplies the magic commit from #852 to making rendering work again.
2. Qt elements were flickering... they no longer are.
3. Python libraries wasn't being detected automatically. The solution in the commit is terrible but works. This is trying to solve this issue: #994 
4. Fixes a crash when closing the game window using the red x button (again on macs). A deconstructor had this instruction in its method body: [ud2](https://hjlebbink.github.io/x86doc/html/UD2.html)

### CODE REVIEW

File: buildsystem/modules/FindPython.cmake [hunk 1]
```diff
@@ -169,6 +170,19 @@ foreach(PYTHON ${PYTHON_INTERPRETERS})
...
+        # On macOS, the above was not working for homebrew installations.
+        # Insetad we inqurie `python3-config --prefix` for the installation,
+        # and find the location of the libraries.
+        if (PYTHON_LIBRARIES STREQUAL "PYTHON_LIBRARIES-NOTFOUND")
+            py_exec(
+                "from distutils.sysconfig import PREFIX; print(PREFIX)"
+                PYTHON_CONFIG_DIR
+            )
+            # TODO: Don't hardcode version numbers
+            set(PYTHON_LIBRARIES "${PYTHON_CONFIG_DIR}/lib/libpython3.6.dylib")
```
Comment (TheJJ) [inline buildsystem/modules/FindPython.cmake]: Please fix the indentation of this block (with tabs).
Why is the hardcoding neccessary?
I would suggest to investigate why `${PYTHON_LIBRARY_NAME}` is not found in `PYTHON_LIBRARY_DIR` where python distutils sysconfig tells us about.

 (also, typos in `insetad`, `inqurie`).

File: libopenage/game_control.h [hunk 2]
```diff
@@ -56,6 +56,7 @@ class OutputModeSignals : public QObject {
 class OutputMode : public input::InputContext {
 public:
 	explicit OutputMode(qtsdl::GuiItemLink *gui_link);
+	virtual ~OutputMode() = 0;
```
Comment (TheJJ) [inline libopenage/game_control.h]: setting it as a pure virtual function is not right. => remove the `= 0`

File: buildsystem/modules/FindPython.cmake [hunk 3]
```diff
@@ -169,6 +170,19 @@ foreach(PYTHON ${PYTHON_INTERPRETERS})
...
+        endif()
+        message("!!! Using ${PYTHON_LIBRARIES}")
```
Comment (TheJJ) [inline buildsystem/modules/FindPython.cmake]: this is likely a leftover.

File: buildsystem/modules/FindPython.cmake [hunk 4]
```diff
@@ -116,6 +116,7 @@ else()
 		"/usr/local/Frameworks/Python.framework/Versions/*/bin/python*"
 		"~/Library/Frameworks/Python.framework/Versions/*/bin/python*"
 		"/System/Library/Frameworks/Python.framework/Versions/*/bin/python*"
+		"/usr/local/opt/python/Frameworks/Versions/*/bin/python*"
```
Comment (TheJJ) [inline buildsystem/modules/FindPython.cmake]: That line should be the only thing needed to find python and its libs correctly, I hope.

Comment (Mrwerdo) [reply buildsystem/modules/FindPython.cmake]: Turns out this is pointless - both `/usr/local/Frameworks/Python.framework/Versions` and `/usr/local/opt/python/Frameworks/Versions` symlink to the same place.

**Comment by Mrwerdo** [comment]
What's holding this up? Were you going to merge this?

**Comment by TheJJ** [comment]
Ah, good. Please do a rebase in order to get rid of the merge commits. In this step you should also meld "Removed erroneous space" into "Ignore macOS .DS_store" and the copyright update commit into the commit that actually changed the contents. I.e. please clean up the history, then we can merge.

**Pull request merged**